app/services/submission_services/download_files.rb

```diff
@@ -0,0 +1,45 @@
+module SubmissionServices
+  class DownloadFiles
+    include Mandate
+    def initialize(submission_uuid, filenames)
+      @filenames = filenames
+      @files = Concurrent::Map.new
+      @path = "#{Rails.env}/storage/#{submission_uuid}"
+    end
+
+    def call
+      threads = []
+      filenames.each do |original_filename, stored_filename|
+        threads << Thread.new { download_file(original_filename, stored_filename) }
+      end
+      threads.each(&:join)
+
+      # Convert to a normal hash for the return.
+      # There's probably a nicer way to do this but
+      # RubyDoc is down for me so :shrug:
```

Comment:
The official documentation can be found https://docs.ruby-lang.org/en/ instead of RubyDoc.  The documentation can also be generated from the source locally.

Reply:
@kotp Thank you.